### DISCUSSION TITLE
Guidance Needed for Modifying File Metadata Across Multiple Batches

### DISCUSSION THREAD

**Discussion by datta202** [rpldy/react-uploady#744] [Q&A]
Hello,

We want to modify the file metadata before the upload starts. Specifically, we need to modify the metadata for multiple batches of files.

Here’s how we envision the process:

1. Users will select files in the uploader and can also remove files from the list after selecting them.
2. Users can select files multiple times. Each selection of files is considered a separate batch. For example, if a user selects 2 files first and then selects 3 more files later, this will result in 2 batches.
3. Now we want to do some checking here and then modify the metadata and then upload to s3

We are uploading files directly to S3.

Currently, we cannot modify the file metadata in the useRequestPreSend hook. I am unsure whether withRequestPreSendUpdate or withBatchStartUpdate would be suitable, as they seem to support only single batches. We need a solution that works for multiple batches.

Could you please help or guide us on how to achieve this?

Thanks!

**Comment by yoavniran** [comment]
hey @datta202 

First, i trust you're already disabling the _autoUpload_ param so user can choose/remove before committing to upload.

As for your main question, Uploady's largest context is of a batch and therefore, there isnt a way to run once over all files that were added across multiple batches. 

I would need to investigate whether it would be possible to support this in the future, but since its hardly ever requested or typically needed, it will most likely remain low priority.

But, I would like to understand why _withBatchStartUpdate_ for example isn't suitable still for your purposes? What is it that you cannot accomplish with it together with some outside state to keep track of any cross-batch data?
Whats the metadata you're looking to modify?

**Reply by datta202** [reply]
Sorry for replying late.

I want to rename multiple batch files before uploading them to S3.

I managed to achieve this by using the useRequestPreSend hook (this hook is called for each file before sending it to S3). I was looking for a hook or a way to rename all files at once and then send them to S3.
